content/upcoming/announcing_babelfish.md

```diff
@@ -0,0 +1,82 @@
+---
+author: abeaumont
+date: 2017-06-06
+title: "Announcing Babelfish"
+draft: true
+image: **TODO: Babelfish image (/post/announcing-babelfish/babelfish.jpg)**
+description: "Announcing Babelfish, the project we are developing to build
+representations of source code."
+categories: ["technical"]
+---
+
+At source{d} we believe there's a possibility for programs to write their own
+code and we are exploring that possibility. Our
+[Roadmap](https://blog.sourced.tech/post/our-roadmap/) states the first step in this
+direction as: *Build representations of source code, developers and projects.*
+
+Today we are announcing Babelfish, the project we are developing to build this
+representations of source code.
+
+## What's Babelfish?
+
+Babelfish is a universal code parser. It can parse any file, in any language,
+extract an abstract syntax tree (AST) from it, and convert it to a Universal
+Abstract Syntax Tree [(UAST)](https://doc.bblf.sh/uast/specification.html)
+
+Well, that's our objective, we are not there just yet. We have designed an
+architecture we're currently working on to achieve that.
+
+## The architecture of Babelfish
+
+**TODO: Architecture image**
+
+As shown in the figure, Babelfish is a client/server application. A client is a
+language analysis tool that wants to perform some analysis on a source code. It
+connects to the server handle source code parsing by using the needed language
```

Comment:
`s/It connects to the server handle source code/It connects to the server to handle source code`?
`s/It connects to the server handle source code/It connects to the server to request parsing of a source code file`?